### PR TITLE
add route rule

### PR DESCRIPTION
**Pull request by john-deng**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,5 +19,5 @@ app.rsa
 
 .idea/
 vendor/
-
+pkg/ci/impl/config/
```
Comment (john-deng) [inline .gitignore]: Not necessary, remove it.

File: pkg/orch/istio/client.go [hunk 2]
```diff
@@ -50,55 +59,20 @@ func newClient(kubeconfig string) (*crd.Client, error) {
...
-	return resourceVersion, nil
+func (client *Client) getConfig() (*model.Config, error) {
+	return nil, nil
 }
 
-func (client *Client) Get() (*model.Config, bool) {
-	config, flag := client.crd.Get(Type, client.Name, client.Namespace)
-	log.Debug("route rule get config", flag)
-	return config, flag
+func (client *Client) Get(typ string) (*model.Config, bool) {
```
Comment (john-deng) [inline pkg/orch/istio/client.go]: func (c *Client) Get(typ string) (*model.Config, bool)

File: pkg/orch/istio/client_test.go [hunk 3]
```diff
@@ -4,71 +4,35 @@ import (
...
-	log.Info("create rule:", resourceVersion)
-}
+
 
 func TestGet(t *testing.T) {
 	config, err := NewClient()
 	assert.Equal(t, nil, err)
-	client := Client{
-		Name:      "hello-world",
+	rule := Client{
```
Comment (john-deng) [inline pkg/orch/istio/client_test.go]: client := Client{

File: pkg/orch/istio/client_test.go [hunk 4]
```diff
@@ -4,71 +4,35 @@ import (
...
+		Name:      "demo-provider",
 		Namespace: "demo-dev",
 	}
-	client.crd =config
-	con, flag := client.Get()
+	rule.Crd = config
+	typ := RouterType
+	con, flag := rule.Get(typ)
```
Comment (john-deng) [inline pkg/orch/istio/client_test.go]: c, flag := client.Get(typ)

File: pkg/orch/istio/client_test.go [hunk 5]
```diff
@@ -4,71 +4,35 @@ import (
...
-			{Weight: 20, Labels: map[string]string{"version": "v2"}},
-		},
-	}
-	config, err := NewClient()
-	client.crd =config
-	assert.Equal(t, nil, err)
-	resourceVersion, err := client.Update()
-	assert.Equal(t, nil, err)
-	log.Info("update rule:", resourceVersion)
+	log.Info("route Rule get version :", resoureceVersion)
```
Comment (john-deng) [inline pkg/orch/istio/client_test.go]: log.Info("routerule get version :", resoureceVersion)

File: pkg/orch/istio/client_test.go [hunk 6]
```diff
@@ -4,71 +4,35 @@ import (
...
 }
 
 func TestDelete(t *testing.T) {
 	config, err := NewClient()
 	assert.Equal(t, nil, err)
-	client := Client{
-		Name:      "hello-world",
+	rule := Client{
```
Comment (john-deng) [inline pkg/orch/istio/client_test.go]: client := Client{

File: pkg/orch/istio/client_test.go [hunk 7]
```diff
@@ -4,71 +4,35 @@ import (
...
+		Name:      "demo-provider",
```
Comment (john-deng) [inline pkg/orch/istio/client_test.go]: Name:      "hello-world",

File: pkg/orch/istio/routerule.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+	Client
+	Route          []*routing.DestinationWeight `json:"route"`
+	HttpFault      *routing.HTTPFaultInjection  `json:"http_fault"`
+	HttpReqTimeout *routing.HTTPTimeout         `json:"http_req_timeout"`
+	Timeout        int64                        `json:"timeout"`
+	Percent        float32                      `json:"percent"`
+	FixedDelay     int64                        `json:"fixed_delay"`
+}
+//Obtain  config
+func (router *RouterRule) GetConfig() (model.Config, error) {
```
Comment (john-deng) [inline pkg/orch/istio/routerule.go]: func (rr *RouterRule) GetConfig() (model.Config, error) {

File: pkg/orch/istio/routerule.go [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+	route := &routing.RouteRule{
```
Comment (john-deng) [inline pkg/orch/istio/routerule.go]: routerule := &routing.RouteRule{

**Review by clarechu — COMMENTED** [review]
aa

**Comment by john-deng** [comment]
close

**Pull request closed without merging**